src/gpg_challenge.c

```diff
@@ -13,55 +13,68 @@
  * limitations under the License.
  */
 
-#include "os.h"
-#include "cx.h"
-#include "gpg_types.h"
-#include "gpg_api.h"
 #include "gpg_vars.h"
+#include "cx_errors.h"
 
 int gpg_apdu_get_challenge() {
-  unsigned int olen, hlen;
+    unsigned int olen, hlen;
+    cx_err_t error = CX_INTERNAL_ERROR;
 
-  if ((G_gpg_vstate.io_p1 & 0x80) == 0x80) {
-    olen = G_gpg_vstate.io_p2;
-  } else {
-    olen = G_gpg_vstate.io_le;
-  }
-  if (olen == 0 || olen > GPG_EXT_CHALLENGE_LENTH) {
-    THROW(SW_WRONG_LENGTH);
-    return SW_WRONG_LENGTH;
-  }
+    if ((G_gpg_vstate.io_p1 & 0x80) == 0x80) {
+        olen = G_gpg_vstate.io_p2;
+    } else {
+        olen = G_gpg_vstate.io_le;
+    }
+    if (olen == 0 || olen > GPG_EXT_CHALLENGE_LENTH) {
+        return SW_WRONG_LENGTH;
+    }
 
-  if ((G_gpg_vstate.io_p1 & 0x82) == 0x82) {
-    unsigned int  path[2];
-    unsigned char chain[32];
-    unsigned char Sr[32];
+    if ((G_gpg_vstate.io_p1 & 0x82) == 0x82) {
+        unsigned int path[2];
+        unsigned char chain[32];
+        unsigned char Sr[64];
```

Comment:
This won't be cleared from memory. Apps should explicitly bzero this.

Reply:
if `memset` is sufficient, then it is done!

Reply:
Should be explicit_bzero to avoid compileur optimisations: https://github.com/LedgerHQ/ledger-secure-sdk/blob/master/lib_standard_app/crypto_helpers.c#L93

Reply:
Done!